lib/AccessoryManager.js

```diff
@@ -500,6 +507,200 @@ class AccessoryManager {
+    const onCharacteristicZwaveNodeValueCriteria = {
+      commandClass: COMMAND_CLASS_SWITCH_MULTILEVEL,
+      valueIndex: SWITCH_BINARY_INDEX_SWITCH
+    }
+
+     const onCharacteristicValueSet = value => {
+       var setValue = brightnessCharacteristic.value
+       if (value) {
+         if (setValue === 1){
+           setValue = 99
+         }
+         else if (setValue === 100) {
+           setValue = 99
+         }
+       } else {
+         setValue = 0
+       }
+       return setValue
+     }
+
+    onCharacteristic
+      .on('get', this._makeCharacteristicGetterForZwaveNodeValue(
+        onCharacteristicName,
+        this._makeZwaveNodeValueResolver(onCharacteristicZwaveNodeValueCriteria),
+        Boolean
+      ))
+      .on('set', this._makeCharacteristicSetterForZwaveNodeValue(
+        onCharacteristicName,
+        onCharacteristicZwaveNodeValueCriteria,
+        onCharacteristicValueSet
+      ))
+
+      // Setup handler for when the value for the "Bightness" characteristic is requested by HomeKit
+      const brightnessCharacteristicName = 'Bightness'
+      const brightnessCharacteristicZwaveNodeValueCriteria = {
+        commandClass: COMMAND_CLASS_SWITCH_MULTILEVEL,
+        valueIndex: SWITCH_BINARY_INDEX_SWITCH
+      }
+
+      brightnessCharacteristic
+        .on('get', this._makeCharacteristicGetterForZwaveNodeValue(
+          brightnessCharacteristicName,
+          this._makeZwaveNodeValueResolver(onCharacteristicZwaveNodeValueCriteria),
+          Number
+        ))
+        .on('set', this._makeCharacteristicSetterForZwaveNodeValue(
+          brightnessCharacteristicName,
+          brightnessCharacteristicZwaveNodeValueCriteria,
+          Number
+        ))
+  }
+  /**
+   * Configure the "ColorLight" service for the accessory
+   */
+   _configureColorLightService () {
+     const lightBulbService = this._getService(this._hapService.Lightbulb)
+     const onCharacteristic = lightBulbService.getCharacteristic(this._hapCharacteristic.On)
+     const hueCharacteristic = lightBulbService.getCharacteristic(this._hapCharacteristic.Hue)
+     const saturationCharacteristic = lightBulbService.getCharacteristic(this._hapCharacteristic.Saturation)
+     const brightnessCharacteristic = lightBulbService.getCharacteristic(this._hapCharacteristic.Brightness)
+
+     // Setup handler for when the value for the "On" characteristic is requested / set by HomeKit
+     const onCharacteristicName = 'On'
+     const onCharacteristicZwaveNodeValueCriteria = {
+       commandClass: COMMAND_CLASS_SWITCH_MULTILEVEL,
+       valueIndex: SWITCH_BINARY_INDEX_SWITCH
+     }
+
+     const onCharacteristicValueSet = value => {
+       var setValue = brightnessCharacteristic.value
+       if (value) {
+         if (setValue === 1){
+           setValue = 99
+         }
+         else if (setValue === 100) {
+           setValue = 99
+         }
+       } else {
+         setValue = 0
+       }
+       return setValue
```

Comment:
Shouldn't this be returning a `boolean` value rather than a `number`? (On value is normally `true` or `false`)